pgmpy/inference/continuous/sampling.py

```diff
@@ -591,3 +592,231 @@ def generate_sample(self, initial_pos, num_adapt, num_samples, trajectory_length
+                    position_bar, candidate_set_size, accept_set_bool)
+
+    def _sample(self, position, stepsize):
+        """
+        Returns a sample using a single iteration of NUTS
+        """
+
+        # Re-sampling momentum
+        momentum = np.random.normal(0, 1, len(position))
+
+        # Initializations
+        depth = 0
+        position_backward, position_forward = position, position
+        momentum_backward, momentum_forward = momentum, momentum
+        candidate_set_size = accept_set_bool = 1
+        _, log_pdf = self.model.get_gradient_log_pdf(position, self.grad_log_pdf)
+
+        # Resample slice variable `u`
+        slice_var = np.random.uniform(0, np.exp(log_pdf - 0.5 * np.dot(momentum, momentum)))
+
+        while accept_set_bool == 1:
+            direction = np.random.choice([-1, 1], p=[0.5, 0.5])
+            if direction == -1:
+                # Build a tree in backward direction
+                (position_backward, momentum_backward, _, _, position_bar,
+                 candidate_set_size2, accept_set_bool2) = self._build_tree(position_backward, momentum_backward,
+                                                                           slice_var, direction, depth, stepsize)
+            else:
+                # Build tree in forward direction
+                (_, _, position_forward, momentum_forward, position_bar,
+                 candidate_set_size2, accept_set_bool2) = self._build_tree(position_forward, momentum_forward,
+                                                                           slice_var, direction, depth, stepsize)
+            if accept_set_bool2 == 1:
+                if np.random.rand() < candidate_set_size2 / candidate_set_size :
+                    position = position_bar
+
+            candidate_set_size += candidate_set_size2
+
+            criteria1 = np.dot((position_forward - position_backward), momentum_backward)>=0
+            criteria2 = np.dot((position_forward - position_backward), momentum_forward)>=0
+            accept_set_bool = accept_set_bool2 and criteria1 and criteria2
+            depth += 1
+
+        return position
+
+    def sample(self, initial_pos, num_samples,stepsize=None):
+        """
+        """
+        initial_pos = _check_1d_array_object(initial_pos, 'initial_pos')
+        _check_length_equal(initial_pos, self.model.variables, 'initial_pos', 'model.variables')
+
+        if stepsize is None:
+            stepsize = self._find_reasonable_stepsize(initial_pos)
+
+        types = [(var_name, 'float') for var_name in self.model.variables]
+        samples = np.zeros(num_samples, dtype=types).view(np.recarray)
+
+        samples[0] = tuple(initial_pos)
+        position_m = initial_pos
+
+        for i in range(1, num_samples):
+            # Genrating sample
+            position_m = self._sample(position_m, stepsize)
+            samples[i] = position_m
+
+        if HAS_PANDAS == True:
+            return pd.DataFrame.from_records(samples)
```

Comment:
@khalibartan There could be case when the user would want a np.recarry output despite having pandas installed. May be we can have an optional parameter to ensure this. What do you think?